src/common/proc_macro/src/session_config.rs

```diff
@@ -292,12 +300,24 @@ pub(crate) fn derive_config(input: DeriveInput) -> TokenStream {
                 }
             }
 
-            /// Show all parameters.
+            /// Show all parameters except those specified `NO_SHOW_ALL`.
             pub fn show_all(&self) -> Vec<VariableInfo> {
                 vec![
                     #(#show_all_list)*
                 ]
             }
+
+            /// List all parameters
+            pub fn list_all(&self) -> Vec<VariableInfo> {
```

Comment:
The difference of these two methods is subtle. Is `NO_SHOW_ALL` essentially `hidden`? 😄

Reply:
BTW, which set of parameters should be attributed with `NO_SHOW_ALL` and why do they need this? Is there any documentation from Postgres that we can refer to?

Reply:
@TennyZhuang